### DISCUSSION TITLE
New Webhook subscription ~

### DISCUSSION THREAD

**Discussion by forwells** [Kyon147/laravel-shopify#282] [General]
I would like to know if after adding a new webhook, it can take effect automatically without reinstalling the app. Or, similarly, I can click Update app in the shopify background to complete the webhook installation.

**Comment by forwells** [comment]
I will use webhook api to dynamic manage webhooks for shopify stores~

**Comment by booni3** [comment]
You can do something like this:

```php
class SyncWebhooksAction
{
    public function execute(User $shop): void
    {
        $shopId = ShopId::fromNative($shop->id);
        $webhooks = config('shopify-app.webhooks');
        WebhookInstaller::dispatch($shopId, $webhooks);
    }
}
```
